openfl/component/aggregator/straggler_handling.py

```diff
@@ -140,3 +192,71 @@ def __minimum_collaborators_reported(self, num_collaborators_done) -> bool:
                 False otherwise.
         """
         return num_collaborators_done >= self.minimum_reporting
+
+
+class PercentagePolicy(StragglerHandlingPolicy):
+    """Percentage based Straggler Handling function."""
+
+    def __init__(self, percent_collaborators_needed=1.0, minimum_reporting=1, **kwargs):
+        """Initialize a PercentagePolicy object.
+
+        Args:
+            percent_collaborators_needed (float, optional): The percentage of
+                collaborators needed. Defaults to 1.0.
+            minimum_reporting (int, optional): The minimum number of
+                collaborators that should report. Defaults to 1.
+            **kwargs: Variable length argument list.
+        """
+        if minimum_reporting <= 0:
+            raise ValueError("minimum_reporting must be >0")
+
+        self.percent_collaborators_needed = percent_collaborators_needed
+        self.minimum_reporting = minimum_reporting
+        self.logger = getLogger(__name__)
```

Comment:
Can we start defining `logger = getLogger(__name__)` below imports and use it globally within this file?
We should slowly move away from these tiny design choices.

Reply:
Accepted. Will incorporate